### PR TITLE
fix(platform): orientation detected correctly

### PR DESCRIPTION
**Pull request by jgw96**
#### Short description of what this resolves:
Orientation is now detected correctly along with the correct window measurements being cached. `window.screen` does not work according to the spec in `uiWebView` and always returns the portrait dimensions even if the device is in landscape, so this was replaced with `win['innerWidth']` and `win['innerHeight']`. There is also extra checking now to keep `_isPortrait` from being set to null in certain cases, such as when the keyboard is up. It now keeps `_isPortrait` correct even when the keyboard is up.

#### Changes proposed in this pull request:

- only check if `win.screen.width` and `win.screen.height` are bigger than 0, do not rely on that api beyond that
- add tests for orientation

**Ionic Version**: 2.x

**Fixes**: #9727 and #10015

### CODE REVIEW

File: src/platform/test/platform.spec.ts [hunk 1]
```diff
@@ -64,6 +64,30 @@ describe('Platform', () => {
...
+
+    it('Should return false if orientation is not portrait', () => {
+      expect(plt.isPortrait()).toEqual(false);
+    });
+
+    it('Should return a number that is equal to window.innerWidth', () => {
+      let type = typeof plt.width();
+
+      expect(type).toEqual('number');
+      expect(plt.width()).toEqual(window.innerWidth);
```
Comment (brandyscarney) [inline src/platform/test/platform.spec.ts]: Could we add some tests that manipulate width & height and check if it is portrait?

**Comment by brandyscarney** [comment]
I confirmed that this issue is fixed on iOS when rotating the screen. I want to do some more testing with the keyboard though. Added a comment for more unit tests - could we add some manipulated width/height and a keyboard if possible? Just to make sure the correct values are being returned.

**Comment by jgw96** [comment]
Definitely down for adding more tests. Need to figure out the best way to manipulate the width and height well.

File: src/platform/platform.ts [hunk 2]
```diff
@@ -571,26 +571,38 @@ export class Platform {
...
+          if (this._lH <= win['innerHeight']) {
+            console.debug('setting _isPortrait to false');
             this._isPortrait = false;
             this._lH = win['innerHeight'];
           }
+          if (this._lW > win['innerWidth']) {
+            // Special case: keyboard is open and device is in portrait
+            console.debug('setting _isPortrait to true while keyboard is open and device is portrait');
+            this._isPortrait = true;
+            this._lW = win['innerWidth'];
```
Comment (brandyscarney) [inline src/platform/platform.ts]: Why are we changing `this._lW` here? If you set it to portrait it is going to grab `this._pW`. I haven't tested this code at all, just glancing at it.

**Comment by numerized** [comment]
Cool !!! Was waiting for this one deadly ... ;)

Patience always works with you guys. App in production couldn't be updated because of this one. 

I'll try it soon.

**Comment by rigothedev** [comment]
Is this ready to be merged?

**Comment by brandyscarney** [comment]
@rigothedev No, we are still testing this. If you'd like to try it out sooner, I've just released a nightly with this PR in it:

```
npm install --save --save-exact ionic-angular@2.0.0-rc.5-201701201813
```

File: src/platform/test/platform.spec.ts (hunk 1 above)
Comment (jgw96) [inline src/platform/test/platform.spec.ts]: Just added some tests for this in this commit https://github.com/driftyco/ionic/pull/10084/commits/d24435f212677cbbc5b9d576dedabc3530f1ff5e

File: src/platform/platform.ts (hunk 2 above)
Comment (jgw96) [inline src/platform/platform.ts]: Looking into this now.

**Comment by numerized** [comment]
@jgw96 From nightly, problem solved.
Thanks a lot.

**Comment by brandyscarney** [comment]
New nightly released with the latest changes: `ionic-angular@2.0.0-rc.5-201701232249`

**Comment by brandyscarney** [comment]
This looks good to me now. I approve, I approve, I approve. 👍 🎉

**Pull request merged**